src/Core/LanguageService.fs

```diff
@@ -1004,3 +1004,14 @@ Consider:
             | Some cl -> return! cl.stop ()
             | None -> return ()
         }
+
+    let activate (context: ExtensionContext) =
+        let restart () =
+            promise {
+                logger.Debug("Restarting F# langugae service")
+                do! stop ()
+                do! start context
+            }
```

Comment:
Is this enough to call `start context`?

Is there no cache to be cleaned like the list of errors, diagnostics, etc?

Reply:
I don't think this is going to be enough - the Project loading chain in `fsharp.fs` is a key part of managing the overall extension state. As currently implemented, the language service would get restarted, but the workspacepeek/workspaceload/projectload dance wouldn't be done - so the newly-started language server would not have any context to begin doing any useful work.

Reply:
Calling the `initWorkspace` function seems to get everything working.